packages/waku/src/lib/renderers/rsc-renderer.ts

```diff
@@ -194,13 +194,14 @@ export async function renderRsc(
 
   let decodedBody: unknown | undefined = args.decodedBody;
   if (body) {
-    const bodyStr = await streamToString(body);
+    const bodyBuf = await streamToArrayBuffer(body);
+    const bodyStr = arrayBufferToString(bodyBuf);
```

Comment:
We can delay calling this in the else-if block. We just need to check if the buffer has a length.

Reply:
great, latest version should have this

Reply:
Cool, thanks.